lib/Notifier.js

```diff
@@ -0,0 +1,371 @@
+   * @param {string} contentType - Content-Type header value.
+   * @param {NotifierOptions} options - Optional parameters.
+   *   @param {Array<string>}  extraHeaders - Additional SIP headers.
+   *   @param {string} allowEvents - Allow-Events header value.
+   *   @param {boolean} pending - Set initial dialog state as "pending". 
+   */
+  constructor(ua, subscribe, contentType, { extraHeaders, allowEvents, pending }) 
+  {
+    logger.debug('new');
+
+    super();
+
+    if (!subscribe)
+    {
+      throw new TypeError('subscribe is undefined');
+    }
+
+    if (!contentType)
+    {
+      throw new TypeError('contentType is undefined');
+    }
+
+    this._ua = ua;
+    this._initial_subscribe = subscribe;
+    this._expires_timestamp = null;
+    this._expires_timer = null;
+
+    // Notifier state: pending, active, terminated. Not used: init, resp_wait.
+    this._state = pending ? C.STATE_PENDING : C.STATE_ACTIVE;
+
+    // Optional. Used to build terminated Subscription-State.
+    this._terminated_reason = null;
+    this._terminated_retry_after = null;
+
+    // Custom session empty object for high level use.
+    this.data = {};
+
+    this._dialog = null;
+    
+    const eventName = subscribe.getHeader('event');
+
+    this._content_type = contentType;
+    this._expires = parseInt(subscribe.getHeader('expires'));
+    this._headers = Utils.cloneArray(extraHeaders);
+    this._headers.push(`Event: ${eventName}`);
+
+    // Use contact from extraHeaders or create it.
+    this._contact = this._headers.find((header) => header.startsWith('Contact'));
+    if (!this._contact)
+    {
+      this._contact = `Contact: ${this._ua._contact.toString()}`;
+
+      this._headers.push(this._contact);
+    }
+
+    if (allowEvents)
+    {
+      this._headers.push(`Allow-Events: ${allowEvents}`);
+    }
+
+    this._target = subscribe.from.uri.user;
+    subscribe.to_tag = Utils.newTag();
+
+    // Create dialog for normal and fetch-subscribe.
+    const dialog = new Dialog(this, subscribe, 'UAS');
+
+    if (dialog.error)
+    {
+      logger.warn(dialog.error);
+
+      throw new Error('SUBSCRIBE missed Contact');
+    }
+      
+    this._dialog = dialog;  
+
+    if (this._expires > 0)
+    {
+      // Set expires timer and time-stamp.
+      this._setExpiresTimer();
+    }
+  }
+
+  /**
+   * Dialog callback.
+   * Called also for initial subscribe. 
+   * Supported RFC 6665 4.4.3: initial fetch subscribe (with expires: 0).
+   */
+  receiveRequest(request) 
+  {
+    if (request.method !== JsSIP_C.SUBSCRIBE) 
+    {
+      request.reply(405);   
+
+      return;
+    }
+
+    let expiresValue = request.getHeader('expires');
+
+    if (expiresValue === undefined || expiresValue === null) 
+    { 
+      // Missed header Expires. RFC 6665 3.1.1. Set default expires value.  
+      expiresValue = '900';
+      logger.debug(`Missed expires header. Set by default ${expiresValue}`);
+    }
+
+    this._expires = parseInt(expiresValue);
```

Comment:
Cosmetic/Readability:

```suggestion    
    if (request.hasHeader('expires')) 
    { 
      this._expires = request.getHeader('expires')
    }
    else
    {    
      // RFC 6665 3.1.1, default expires value.
      this._expires = 900;
      
      logger.debug(`missing Expires header field, default value set: ${this._expires}`);
    }
```

Reply:
OK, 

One note - getHeader returns string, not number
So better use this._expires = parseInt(request.getHeader('expires'))